src/test/java/com/qunite/api/service/UserServiceTest.java

```diff
@@ -41,6 +43,31 @@ void cleanAll() {
     entryRepository.deleteAll();
   }
 
+  @Test
+  @Sql("/users-create.sql")
+  void testUpdatingShouldUpdateUserWithValidData() {
+    var user = userService.findOne(1L).get();
+    user.setUsername("NewUsername");
+
+    userService.updateOne(user);
+
+    var realUsername = userService.findOne(1L).get().getUsername();
+    assertThat(realUsername).isEqualTo("NewUsername");
+  }
+
+  @Test
+  @Sql("/users-create.sql")
+  void testUpdatingShouldNotUpdateUserWithUsedUsernameOrEmail() {
+    var user = userService.findOne(1L).get();
+    user.setUsername("Second");
+    user.setEmail("User2@user.com");
+
+    Exception exception = assertThrows(UserAlreadyExistsException.class, () ->
+        userService.updateOne(user));
+
+    assertThat(exception.getMessage()).isEqualTo("Username or email is already in use");
+  }
```

Comment:
1. Agree

Reply:
2. do you really think that smth like this would be more readable and better?
![image](https://github.com/q-unite/qunite-api/assets/103951082/647cafae-80b0-4fe5-9213-caa8963cd8a4)

Reply:
1. Not sure, need to check where replacing service with repository is appropriate and where is not, `ll do it later